components/go/pkg/api/feature_entitlement.go

```diff
@@ -0,0 +1,49 @@
+package api
+
+type FeatureEntitlement struct {
+	Entitled   bool            `json:"entitled"`
+	EntitledBy map[string]bool `json:"entitledBy"`
+}
+
+type featureResponse struct {
+	Data       FeatureEntitlement `json:"data"`
+	StatusCode int                `json:"status"`
+}
+
+func (c APIClient) GetFeatureByName(featureName string, keys []string) (*FeatureEntitlement, error) {
+	params := struct {
+		Keys        []string `json:"licenseIds"`
+		FeatureName string   `json:"featureName"`
+	}{
+		Keys:        keys,
+		FeatureName: featureName,
+	}
+
+	resp, err := c.doPOSTRequest("featurebyname", params)
+	if err != nil {
+		return nil, err
+	}
+
+	var data featureResponse
+	c.decodeJSON(resp, &data)
+	return &data.Data, nil
+}
+
+func (c APIClient) GetFeatureByGUID(featureID string, keys []string) (*FeatureEntitlement, error) {
+	params := struct {
+		Keys      []string `json:"licenseIds"`
+		FeatureID string   `json:"featureGuid"`
+	}{
+		Keys:      keys,
+		FeatureID: featureID,
+	}
+
+	resp, err := c.doPOSTRequest("featurebyid", params)
+	if err != nil {
+		return nil, err
+	}
+
+	var data featureResponse
+	c.decodeJSON(resp, &data)
```

Comment:
require to handle error if failed to decode JSON response and other places also

Reply:
I've addressed the error handling on the decodeJSON function itself.